src/test/java/org/gridsuite/modification/server/ModificationControllerTest.java

```diff
@@ -224,6 +224,68 @@ public void testSwitch() {
         testDeleteNetwokModifications(TEST_NETWORK_ID, 3);
     }
 
+    @Test
+    public void testLine() {
+        // network not existing
+        webTestClient.put().uri("/v1/networks/{networkUuid}/lines/{lineId}/lockout", NOT_FOUND_NETWORK_ID, "line2")
+                .exchange()
+                .expectStatus().isNotFound()
+                .expectBody(String.class)
+                .isEqualTo(new NetworkModificationException(NETWORK_NOT_FOUND, NOT_FOUND_NETWORK_ID.toString()).getMessage());
+
+        // line not existing
+        webTestClient.put().uri("/v1/networks/{networkUuid}/lines/{lineId}/lockout", TEST_NETWORK_ID, "notFound")
+                .exchange()
+                .expectStatus().isNotFound()
+                .expectBody(String.class)
+                .isEqualTo(new NetworkModificationException(LINE_NOT_FOUND, "notFound").getMessage());
+
+        // line lockout
+        webTestClient.put().uri("/v1/networks/{networkUuid}/lines/{lineId}/lockout", TEST_NETWORK_ID, "line2")
+                .exchange()
+                .expectStatus().isOk()
+                .expectHeader().contentType(MediaType.APPLICATION_JSON)
+                .expectBodyList(ElementaryModificationInfos.class)
+                .value(modifications -> modifications.get(0),
+                        createMatcherElementaryModificationInfos("v1bl1", Set.of("s1"), "open", true));
+
+        // line switch on
+        webTestClient.put().uri("/v1/networks/{networkUuid}/lines/{lineId}/switchOn", TEST_NETWORK_ID, "line2")
+                .exchange()
+                .expectStatus().isOk()
+                .expectHeader().contentType(MediaType.APPLICATION_JSON)
+                .expectBodyList(ElementaryModificationInfos.class)
+                .isEqualTo(List.of());
+
+        // line trip
+        webTestClient.put().uri("/v1/networks/{networkUuid}/lines/{lineId}/trip", TEST_NETWORK_ID, "line2")
+                .exchange()
+                .expectStatus().isOk()
+                .expectHeader().contentType(MediaType.APPLICATION_JSON)
+                .expectBodyList(ElementaryModificationInfos.class)
+                .value(modifications -> modifications.get(0),
+                        createMatcherElementaryModificationInfos("v1bl1", Set.of("s1"), "open", true));
+
+        // line energise on one end
+        webTestClient.put().uri("/v1/networks/{networkUuid}/lines/{lineId}/energiseEnd?side=ONE", TEST_NETWORK_ID, "line2")
+                .exchange()
+                .expectStatus().isOk()
+                .expectHeader().contentType(MediaType.APPLICATION_JSON)
+                .expectBodyList(ElementaryModificationInfos.class)
+                .value(modifications -> modifications.get(0),
+                        createMatcherElementaryModificationInfos("v3bl1", Set.of("s2"), "open", true));
```

Comment:
Same above

Reply:
Tests added

Reply:
Status ?

Reply:
The status of the line is not modified in this case